### PR TITLE
fix(api-client): move web hash router to a factory function

### LINKED ISSUE #2791: Unexpected `/` added to hash
### What happens?

It seems like our change to using the WebRequestRouter in #2597 added a `/` to the beginning of the hash on the navigate which is causing a number of comparisons to fail.

This is making tags not expand / collapse properly and making sidebar items not collapse correctly.

### What did you expect to happen?

There shouldn't be a `/` in the id on navigate.

### How can we reproduce the issue?

Click on any of the sidebar items or tags.

### OpenAPI Document

_No response_

### PR DESCRIPTION
**Pull request by hwkr**
This took me way to long to debug but fixes #2791 and also the issue with sidebar items not highlighting when active.

@hanspagel I tested the app and everything seemed to be working fine still but if you want to confirm that wouldn't hurt 🙏

### CODE REVIEW

**Comment by hwkr** [comment]
@hanspagel @amritk @marclave I wasn't sure if we should also switch the [normal router](https://github.com/scalar/scalar/blob/main/packages/api-client/src/router.ts#L132) to be a factory function since it seems like doing multiple calls to the router history functions may have weird side effects 🤔

**Comment by amritk** [comment]
Hey so we needed the factory function in the modal due to multiple specs on one page, just curious why it was required here?

**Comment by hwkr** [comment]
> Hey so we needed the factory function in the modal due to multiple specs on one page, just curious why it was required here?

See https://github.com/scalar/scalar/issues/2791#issuecomment-2272090188

Tl;dr; is calling `createWebHashHistory` has side effects

**Comment by hanspagel** [comment]
Wow, what a journey!

**Review by hanspagel — APPROVED** [review]
Must have been crazy to hunt that bug, I’d watch if they’d have a documentary on Netflix on it. The fix works well!

**Review by hanspagel — CHANGES_REQUESTED** [review]
Oops, ignore this comment here. :)

**Review by hanspagel — APPROVED** [review]

**Review by amritk — APPROVED** [review]
Okay I checked it out, holy what a bug. We should report it to vue-router! Also nice fix

**Review by marclave — APPROVED** [review]
:)
